pytest-embedded-idf/pytest_embedded_idf/unity_tester.py

```diff
@@ -418,27 +418,70 @@ def run_single_board_case(self, name: str, reset: bool = False, timeout: float =
 
     def run_all_single_board_cases(
         self,
-        group: t.Optional[str] = None,
+        group: t.Optional[t.Union[str, list]] = None,
         reset: bool = False,
         timeout: float = 30,
         run_ignore_cases: bool = False,
+        name: t.Optional[t.Union[str, list]] = None,
+        attributes: t.Optional[dict] = None,
     ) -> None:
         """
         Run all single board cases, including multi_stage cases, and normal cases
 
         Args:
-            group: test case group
-            reset: whether to perform a hardware reset before running a case
-            timeout: timeout. (Default: 30 seconds)
-            run_ignore_cases: run ignored test cases or not
+            group: test case group or a list of test case groups to run. Supports 'and' with '&'.
+            Supports group inversion with '!'.
+            reset: Whether to perform a hardware reset before running a test case.
+            timeout: Timeout in seconds. (Default: 30 seconds)
+            run_ignore_cases: Whether to run ignored test cases or not.
+            name: test case name or a list of test case names to run.
+            attributes: Dictionary of attributes to filter and run test cases.
         """
+        if group is None:
+            group = []
+        if isinstance(group, str):
+            group: list[str] = [group]
+        group: list[list[str]] = [[_and.strip() for _and in _or.split('&')] for _or in group]
+
+        if isinstance(name, str):
+            name: list[str] = [name]
+
+        def validate_group(case_groups):
+            if not group:
+                return True
+
+            for _or in group:
+                for _and in _or:
+                    invert = _and.startswith('!')
+                    _and = _and.lstrip('!')
+                    result = _and in case_groups
+                    if invert:
+                        result = not result
+                    if not result:
+                        break
+                else:
+                    return True
+
+            return False
+
         for case in self.test_menu:
-            if not group or group in case.groups:
-                if not case.is_ignored or run_ignore_cases:
-                    if case.type == 'normal':
-                        self._run_normal_case(case, reset=reset, timeout=timeout)
-                    elif case.type == 'multi_stage':
-                        self._run_multi_stage_case(case, reset=reset, timeout=timeout)
+            should_run = False
+            if not group and not name and not attributes:
+                should_run = True
+            if group and validate_group(case.groups):
+                should_run = True
+            if name and case.name in name:
+                should_run = True
+            if attributes and all(case.attributes.get(k) == v for k, v in attributes.items()):
+                should_run = True
```

Comment:
```suggestion
            if group and validate_group(case.groups):
                should_run = True
            if name and case.name in name:
                should_run = True
            if attributes and all(case.attributes.get(k) == v for k, v in attributes.items()):
                should_run = True
```

here the logic is not clear in the documentation.

if the user specified `group`, `name`, and `attributes` all together, test cases that matched "group" or "name" or "attribute" would be run.

I think the behavior is okay, but we could put a `Notes` in the docstring.

Reply:
Yes there is 'or' condition. I will note it